presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/thrift/ThriftHiveMetastore.java

```diff
@@ -343,6 +343,14 @@ public ThriftMetastoreStats getStats()
                     .stopOnIllegalExceptions()
                     .run("getTable", stats.getGetTable().wrap(() -> {
                         Table table = getTableFromMetastore(identity, databaseName, tableName);
+
+                        // Check if hive view is supported
+                        if (!translateHiveViews
```

Comment:
Similar to the other comment, this check should be done in `HiveMetadata` where the translation occurs.